provider/azure/internal/azurecli/az.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package azurecli
+
+import (
+	"encoding/json"
+	"os/exec"
+	"strings"
+
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+)
+
+// Logger for the Azure provider.
+var logger = loggo.GetLogger("juju.provider.azure.internal.azurecli")
+
+// executor is the subset of the exec.Command functionality used by Exec.
+type executor interface {
+	Output() ([]byte, error)
+}
+
+// newExecutor is used to create a new exec.Command for Exec. It is a
+// variable so that it can be overridden in tests.
+var newExecutor = func(command string, args []string) executor {
+	return exec.Command(command, args...)
+}
+
+// Exec attempts to execute "az" with the given arguments. Unmarshalling
+// the json output into v.
+func Exec(v interface{}, args ...string) error {
+	args = append(args, "-o", "json")
+	logger.Debugf("running az %s", strings.Join(args, " "))
+	cmd := newExecutor("az", args)
```

Comment:
Shouldn't we send stderr somewhere other than /dev/null?
If the command fails, "execution failure: exit status 1" won't be very informative.